modules/lang-mustache/src/test/java/org/elasticsearch/script/mustache/MustacheScriptEngineTests.java

```diff
@@ -196,6 +200,33 @@ public void testSimple() throws IOException {
         assertThat(TemplateScript.execute(), equalTo("{\"match_all\":{}}"));
     }
 
+    public void testDetectMissingParam() throws IOException {
```

Comment:
Do we have tests that will verify we correctly function when detect missing param is false AND params are missing/null?